### PR TITLE
fix/13-complete-app-flow

### PR DESCRIPTION
**Pull request by zohaibwebdev**
image

### CODE REVIEW

File: src/shared/components/custom-button/custom-button.types.ts [hunk 1]
```diff
@@ -7,11 +7,13 @@ export interface CustomButtonProps extends PropsWithChildren {
     classNames?: string
     disable?: boolean
     hidden?: boolean
+    buttonType?: 'submit' | 'reset' | 'button'
```
Comment (awaisshahid965) [inline src/shared/components/custom-button/custom-button.types.ts]: should need to remove this

File: src/shared/components/summary/summary-detail/summary-detail.component.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import React, { FC } from 'react'
+import { SummaryDetailProps } from '../summary.types'
+import LoaderIcon from '@/shared/icons/loader.icon'
+
+const SummaryDetail: FC<SummaryDetailProps> = ({ loading, summary, error }) => {
+    return (
+        <div className="my-10 max-w-full flex justify-center items-center">
+            {loading ? (
```
Comment (awaisshahid965) [inline src/shared/components/summary/summary-detail/summary-detail.component.tsx]: we are not allowed to use ternary operator in jsx

File: src/shared/components/summary/summary-form/summary-form.component.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+import CustomButton from '../../custom-button/custom-button.component'
+import { CustomButtonTypes } from '../../custom-button/custom-button.types'
+
+const SummaryForm = () => {
+    const handleInput = () => {}
+
+    const handleSubmit = () => {}
+
+    return (
+        <form action="" className="relative">
```
Comment (awaisshahid965) [inline src/shared/components/summary/summary-form/summary-form.component.tsx]: such case can also be done without forms

File: src/shared/components/summary/summary-history/summary-history.component.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import { FC, useState } from 'react'
+import { HistoryProps } from '../summary.types'
+import CopyIcon from '@/shared/icons/copy.icon'
+import TickIcon from '@/shared/icons/tick.icon'
+
+const SummaryHistory: FC<HistoryProps> = ({ history }) => {
```
Comment (awaisshahid965) [inline src/shared/components/summary/summary-history/summary-history.component.tsx]: what does this summary means?

Comment (awaisshahid965) [inline src/shared/components/summary/summary-history/summary-history.component.tsx]: where is the function that handles what happens when we click on history

Comment (awaisshahid965) [inline src/shared/components/summary/summary-history/summary-history.component.tsx]: plus component name needs to be changed that implies what it is intended to do

File: src/shared/components/summary/summary-history/summary-history.component.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    const [copy, setCopy] = useState('')
+    const handleCopy = (url: string) => {
+        setCopy(url)
```
Comment (awaisshahid965) [inline src/shared/components/summary/summary-history/summary-history.component.tsx]: for this, create a ticket in backlog says "optimize history component"

File: src/shared/components/summary/summary.types.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+export interface HistoryProps {
+    history: Array<string>
+}
+
+export interface SummaryDetailProps {
+    loading?: boolean
```
Comment (awaisshahid965) [inline src/shared/components/summary/summary.types.ts]: why every member is optional? fix this

File: src/shared/icons/tick.icon.tsx [hunk 7]
```diff
@@ -0,0 +1,25 @@
+import React from 'react'
+
+const TickIcon = () => {
```
Comment (awaisshahid965) [inline src/shared/icons/tick.icon.tsx:3]: where is default width and height? or what i need to do if i have to change any of its dimension

File: src/shared/icons/loader.icon.tsx [hunk 8]
```diff
@@ -0,0 +1,196 @@
+import React from 'react'
+
+const LoaderIcon = () => {
```
Comment (awaisshahid965) [inline src/shared/icons/loader.icon.tsx]: same issue as in TickIcon

**Review by awaisshahid965 — CHANGES_REQUESTED** [review]

File: src/shared/components/custom-button/custom-button.types.ts (hunk 1 above)
Comment (zohaibwebdev) [reply src/shared/components/custom-button/custom-button.types.ts]: removed

File: src/shared/components/summary/summary-form/summary-form.component.tsx (hunk 3 above)
Comment (zohaibwebdev) [reply src/shared/components/summary/summary-form/summary-form.component.tsx]: done

**Comment by zohaibwebdev** [comment]
image

**Pull request merged**